docs/guides/collections.rst

```diff
@@ -1,105 +1,144 @@
+instantiated with the following filter, like so:
 
-                    def instantiate(self, name, label):
-                        return Object(self, name, label)
+.. code-block:: python
 
-                    def create(self, name, label):
-                        run_create(name, label)
-                        return self.instantiate(name, label)
+                repo = appliance.collections.repositories.all()[0]
+                playbook_collection = repo.collections.playbooks
+                playbook_collection.all()  # returns ONLY playbooks from that repo
 
-                class Object(BaseEntity):
-                    """An object entity"""
+                # equivalent code
+                playbook_collection = PlaybookCollection(self.appliance, filters={'parent': self})
 
-                    def __init__(self, collection, name, label):
-                        self.collection = collection
-                        self.appliance = self.collection.appliance
+In the example above, the ``BaseEntity`` automatically instantiates the playbook collection object
+with a ``parent`` filter. The playbook collection object would then need to honour that filter
+when returning the playbooks. A collection isn't under any obligation to support a certain filter.
 
-                    def update(self, *updates):
-                        some_update_mechanism(*updates)
+.. note:: In the future filter names which are supported may need to be defined somewhere to allow
+          unsupported filters to be reported as warnings.
+
+Collection Methods
+------------------
+
+* ``__init__()`` - This method is hidden inside the ``BaseCollection`` object and shouldn't be
+  overidden without good reason. There are exceptional circumstances. Mainly if work **MUST** be
+  done at init time, using ``__attrs_post_init__`` method is used instead.
+* ``instantiate()`` - This method is provided by the ``BaseCollection`` and uses the ``ENTITY``
+  attribute of the collection class to determine which class to use in creating the entity.
+* ``create()`` - The collection object **should** provider a ``create()`` method where appropriate.
+  This method will attempt to create the object on the appliance **must** call
+  ``self.instantiate`` to obtain the object to return.
+
+
+Entity Methods
+--------------
+
+* ``__init__()`` - This method is hidden inside the ``BaseEntity`` object and shouldn't be overidden
+  without good reason. There are exceptional circumstances. Mainly if work **MUST** be
+  done at init time, using ``__attrs_post_init__`` method is used instead.
+
+
+Example
+-------
 
-An example of the models usage in testing is described below
+Below is an example of the usage of a collection object described above
 
 .. code-block:: python
 
-                from some.model import ObjectCollection
+                repo = appliance.collections.ansible_repositories.all()[0]
```

Comment:
this `...[0]` patter begs for a better name

for the sake of testing we ought to consider intentionally controlled or random selection of elements